source/common/common/utility.cc

```diff
@@ -122,6 +122,15 @@ std::vector<std::string> StringUtil::split(const std::string& source, const std:
   return ret;
 }
 
+std::string StringUtil::join(const std::vector<std::string>& source, const std::string& delimiter) {
```

Comment:
Maybe add a `TODO` to templatize if you don't think it's necessary in this PR. The first arg could really be a number of things, as long as they can be treated effectively as a `std::string` iterator.